ui/apps/platform/src/Containers/MainPage/Header/CLIDownloadMenu.tsx

```diff
@@ -16,95 +34,61 @@ type CLIDownloadMenuProps = {
 function CLIDownloadMenu({ addToast, removeToast }: CLIDownloadMenuProps): ReactElement {
     const [isCLIMenuOpen, setIsCLIMenuOpen] = useState(false);
 
-    function handleDownloadCLI(os: string) {
+    function handleDownloadCLI(os: AvailableOS) {
         return () => {
+            setIsCLIMenuOpen(false);
+            addToast(`Downloading roxctl for ${os}`);
             downloadCLI(os)
-                .then(() => {
-                    setIsCLIMenuOpen(false);
-                })
                 .catch((err) => {
                     addToast(`Error while downloading roxctl for ${os}`);
                     removeToast();
                     Raven.captureException(err);
+                })
+                .finally(() => {
+                    removeToast();
```

Comment:
I'm not sure that `removeToast()` is working correctly, as it doesn't seem to automatically remove the toast component. These should be ported to PatternFly anyway, so I did not investigate the implementation too deeply.

The toast is still dismiss-able by clicking the X button.